client/app/bundles/course/survey/pages/SurveyShow/Section/__test__/NewQuestionButton.test.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { mount, ReactWrapper } from 'enzyme';
+import ReactTestUtils from 'react-addons-test-utils';
+import CourseAPI from 'api/course';
+import storeCreator from '../../../../store';
+import NewQuestionButton from '../NewQuestionButton';
+import QuestionFormDialogue from '../../../../containers/QuestionFormDialogue';
+
+describe('<NewQuestionButton />', () => {
+  it('injects handlers that allow survey questions to be created', () => {
+    const spyCreate = jest.spyOn(CourseAPI.survey.questions, 'create');
+    const store = storeCreator({ surveys: {} });
+    const contextOptions = {
+      context: { intl, store, muiTheme },
+      childContextTypes: { muiTheme: React.PropTypes.object, intl: intlShape },
+    };
+
+    const sectionId = 7;
+    const newQuestionButton = mount(<NewQuestionButton sectionId={sectionId} />, contextOptions);
+    const questionFormDialogue = mount(<QuestionFormDialogue />, contextOptions);
+
+    // Click 'new question' button
+    const newQuestionButtonNode = ReactDOM.findDOMNode(newQuestionButton.find('button').node);
+    ReactTestUtils.Simulate.touchTap(newQuestionButtonNode);
+    expect(questionFormDialogue.find('QuestionFormDialogue').first().props().visible).toBe(true);
+
+    // Fill section form with title
+    const questionText = 'Question: Is it true?';
+    const optionText = 'Yes';
+    const dialogInline = questionFormDialogue.find('RenderToLayer').first().node.layerElement;
+    const questionForm = new ReactWrapper(dialogInline, true).find('form');
+    const descriptionInput = questionForm.find('textarea[name="description"]');
+    descriptionInput.simulate('change', { target: { value: questionText } });
+    const optionInput = questionForm.find('QuestionFormOption').first().find('textarea').last();
+    optionInput.simulate('change', { target: { value: optionText } });
+
+    // Submit question form
+    const submitButton = questionFormDialogue.find('FormDialogue').first().node.submitButton;
+    ReactTestUtils.Simulate.touchTap(ReactDOM.findDOMNode(submitButton));
+
+    expect(spyCreate).toHaveBeenCalled();
+    const formData = spyCreate.mock.calls[0][0];
+    // Test one by one because jsdom does not support 'FormData#getAll' yet
```

Comment:
Update the comment to
```
// formData is an instance of FormData. To enumerate all the items, we should be able to use
// `#entries` or `#keys`, but jsdom doesn't seem to support these methods yet.
// See https://github.com/tmpvar/jsdom/issues/1671
```
FormData is used because we need to upload files. [This](http://stackoverflow.com/questions/41431322/how-to-convert-formdatahtml5-object-to-json) might help also.